### ISSUE TITLE
Possible to use as submodule and extend with another template?

### ISSUE THREAD

**Issue by opncow** [dirkolbrich/hugo-theme-tailwindcss-starter#13]
I just tried to use hugo-theme-tailwindcss-starter as a Git submodule and extend from it with my own theme. However, I noticed that, for example, the `site.css` in my theme that should override the one that comes with hugo-theme-tailwindcss-starter start was not taken into account.

Do you know of this behavior and can explain it? Can it be fixed?

It would be nice to not lose the connection to the original repo and be able to update it.

**Comment by dirkolbrich** [comment]
Hey,

I tested a basic setup with a combined theme from this packages loaded as a git submodule under `/themes` and a custom theme based on this package. It seems to work.

```bash
// installation process for a new hugo site

➜  gohugo hugo new site test-submodule-theme
➜  gohugo cd test-submodule-theme 
➜  test-submodule-theme git init
➜  test-submodule-theme git:(master) ✗ git add .
➜  test-submodule-theme git:(master) ✗ git commit -m "initial commit"

// load the hugo-theme-tailwindcss-starter package as a submodule under /themes

➜  test-submodule-theme git:(master) git submodule add https://github.com/dirkolbrich/hugo-theme-tailwindcss-starter themes/tailwindcss-starter
➜  test-submodule-theme git:(master) ✗ cd themes/tailwindcss-starter 
➜  tailwindcss-starter git:(master) npm install

// here I add theme = "tailwindcss-starter" to test-submodule-theme/config.toml
// jump back to the root of the site and test if the submodule theme works

➜  tailwindcss-starter git:(master) cd ..
➜  themes git:(master) ✗ cd ..
➜  test-submodule-theme git:(master) ✗ hugo server                  

// now I move back to the /themes folder and install the hugo-theme-tailwindcss-starter package as a separate theme
                                                                                                                                                                                                  
➜  test-submodule-theme git:(master) ✗ cd themes
➜  themes git:(master) ✗ git clone https://github.com/dirkolbrich/hugo-theme-tailwindcss-starter my-theme      
➜  themes git:(master) ✗ cd my-theme 
➜  my-theme git:(master) rm -rf .git
➜  my-theme git:(master) ✗ npm install

// I change the line to theme = ["my-theme", "tailwindcss-starter"] in test-submodule-theme/config.toml to let "my-theme" take precedence over "tailwindcss-starter"
// then I change the the color of the h1 heading in test-submodule-theme/themes/my-theme/layouts/index.html to text-red-500
// jump back to the root of the site and test if my-theme works

➜  my-theme git:(master) ✗ cd ..
➜  themes git:(master) ✗ cd ..
➜  test-submodule-theme git:(master) ✗ Hugo server

// the initial text is now red instead of gray

```

Could you please post your process and any error messages so I can reproduce the issue.

**Comment by dirkolbrich** [comment]
To your initial question: As shown above, using this package as a submodule and extending with another template is possible. But it does not make any sense.

This is a boilerplate starter package for the basic setup of Tailwind CSS. There is basically no other functionality than that. And there probably never will. 

No need to keep this package as a dependency in your project. Any update would just be made to keep up with the Tailwind CSS functionality, and this is covered with a simple `npm update`.

**Comment by opncow** [comment]
Thanks for looking into this. I will try it out again and take notes on what exactly I have done and come back at you.

I will also think twice if I really need the starter as a submodule... thanks!

**Comment by opncow** [comment]
So I have experimented a bit more and hope that I can now better explain what does not work. In contrast to yours, in my version, `my-theme` is not a complete copy of the starter. Instead, I have just copied and adjusted some files from the starter in `my-theme`. 

The Hugo dev server still works, but in production builds (`hugo`), for example, css classes used in `index.html` of `my-theme` are missing.

I'm now experimenting with subtree and modifying the starter directly: https://www.atlassian.com/git/tutorials/git-subtree I'm just not sure, yet, how incorporating updates from the starter will work.

Maybe I will also just leave this aside soon and go by the approach in the docs, given you said that you do not plan to make bigger changes.
